packages/peregrine/lib/talons/CheckoutPage/ItemsReview/useItemsReview.js

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState, useCallback } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
+
+import { useCartContext } from '../../../context/cart';
+
+export const useItemsReview = props => {
+    const [items, setItems] = useState([]);
+    const [totalQuantity, setTotalQuantity] = useState(0);
+    const [showAllItems, setShowAllItems] = useState(false);
+
+    const {
+        queries: { getItemsInCart }
+    } = props;
+
+    const [{ cartId }] = useCartContext();
+
+    const [fetchItemsInCart, { called, data, error, loading }] = useLazyQuery(
+        getItemsInCart,
+        {
+            fetchPolicy: 'cache-and-network'
+        }
+    );
+
+    const setShowAllItemsFlag = useCallback(() => setShowAllItems(true), [
+        setShowAllItems
+    ]);
+
+    useEffect(() => {
+        if (cartId) {
+            fetchItemsInCart({
+                variables: {
+                    cartId
+                }
+            });
+        }
+    }, [cartId, fetchItemsInCart]);
+
+    useEffect(() => {
+        if (error) {
+            console.error(error);
+            setItems([]);
+            setTotalQuantity(0);
+        }
+    }, [error]);
+
+    useEffect(() => {
+        if (called && !error && !loading) {
+            /**
+             * if for some reason items turns out to be
+             * null set items to []
+             */
+            setItems(data.cart.items || []);
+            /**
+             * if total_quantity turns out to be null
+             * + will convert it to 0
+             */
+            setTotalQuantity(+data.cart.total_quantity);
```

Comment:
I don't think we need local state for `items` or `totalQuantity`. Should just be able to return the values from `data` or some default:

```js
const items = data ? data.cart.items : [];
return {
  items: showAllItems ? items : items.slice(0, 2),
  totalQuantity: data ? data.cart.total_quantity : 0
  ...
}

Reply:
Makes sense, data is probably a state variable coming from the apollo hooks. Done.